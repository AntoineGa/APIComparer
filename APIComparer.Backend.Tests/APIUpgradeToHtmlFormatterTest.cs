namespace APIComparer.Tests.VersionComparisons
{
    using System.IO;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using APIComparer.Backend;
    using APIComparer.Backend.Reporting;
    using APIComparer.VersionComparisons;
    using NUnit.Framework;

    [TestFixture]
    [UseReporter(typeof(DiffReporter),typeof(ClipboardReporter))]
    public class APIUpgradeToHtmlFormatterTest
    {
        [Test]
        public void TestComplianceNewtonsoftJson()
        {
            Approvals.VerifyHtml(HtmlDiff("newtonsoft.json", "5.0.8", "6.0.8"));
        }

        [Test]
        public void TestComplianceNServiceBus()
        {
            Approvals.VerifyHtml(HtmlDiff("nservicebus", "4.0.0", "5.0.0"));
        }

        [Test]
        public void TestComplianceAzureSelfDestruct()
        {
            Approvals.VerifyHtml(HtmlDiff("Two10.Azure.SelfDestruct", "1.0.0", "1.0.5"));
        }

        [Test]
        public void TestComplianceLibLog()
        {
            Approvals.VerifyHtml(HtmlDiff("LibLog", "3.0.0", "4.1.1"));
        }

        [Test]
        public void AppccelerateEventBrokerNoChanges()
        {
            Approvals.VerifyHtml(HtmlDiff("Appccelerate.EventBroker", "3.1.0", "3.15.0"));
        }

        private string HtmlDiff(string packageName, string fromVersion, string untilVersion)
        {
            using (var writer = new StringWriter())
            {
                var formatter = new APIUpgradeToHtmlFormatter();

                var packageDescription = new PackageDescription
                {
                    PackageId = packageName,
                    Versions = new VersionPair(fromVersion, untilVersion)
                };

                var compareSetCreator = new CompareSetCreator();
                var sets = compareSetCreator.Create(packageDescription);
                var compareSetDiffer = new CompareSetDiffer();
                var diff = compareSetDiffer.Diff(sets);

                formatter.Render(writer, packageDescription, diff);

                return writer.ToString();
            }
        }
    }
}
