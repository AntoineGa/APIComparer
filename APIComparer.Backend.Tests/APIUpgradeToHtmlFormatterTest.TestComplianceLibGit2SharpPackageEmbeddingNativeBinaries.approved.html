


<h1>.NETFramework,Version=v4.0</h1>

    
        


        


        


        

<h2>The following types have Obsoletes.</h2>

<h3>LibGit2Sharp.GlobalSettings</h3>






<h4>Obsolete Methods</h4>
<ul>
  
  <li><code>LibGit2Sharp.BuiltInFeatures Features()</code><br/> This method will be removed in the next release. Use Version.Features instead. Obsoleted with warning.</li>
  
</ul>



<h3>LibGit2Sharp.ReferenceCollectionExtensions</h3>






<h4>Obsolete Methods</h4>
<ul>
  
  <li><code>LibGit2Sharp.Reference Move(LibGit2Sharp.ReferenceCollection, string, string, LibGit2Sharp.Signature, string, bool)</code><br/> This method will be removed in the next release. Please use Rename() instead. Obsoleted with warning.</li>
  
</ul>



<h3>LibGit2Sharp.Index</h3>






<h4>Obsolete Methods</h4>
<ul>
  
  <li><code>void Stage(string, LibGit2Sharp.StageOptions)</code><br/> This method will be removed in the next release. Use IRepository.Stage() instead. Obsoleted with warning.</li>
  
  <li><code>void Stage(IEnumerable&lt;string&gt;, LibGit2Sharp.StageOptions)</code><br/> This method will be removed in the next release. Use IRepository.Stage() instead. Obsoleted with warning.</li>
  
  <li><code>void Unstage(string, LibGit2Sharp.ExplicitPathsOptions)</code><br/> This method will be removed in the next release. Use IRepository.Unstage() instead. Obsoleted with warning.</li>
  
  <li><code>void Unstage(IEnumerable&lt;string&gt;, LibGit2Sharp.ExplicitPathsOptions)</code><br/> This method will be removed in the next release. Use IRepository.Unstage() instead. Obsoleted with warning.</li>
  
  <li><code>void Move(string, string)</code><br/> This method will be removed in the next release. Use IRepository.Move() instead. Obsoleted with warning.</li>
  
  <li><code>void Move(IEnumerable&lt;string&gt;, IEnumerable&lt;string&gt;)</code><br/> This method will be removed in the next release. Use IRepository.Move() instead. Obsoleted with warning.</li>
  
  <li><code>void Remove(string, bool, LibGit2Sharp.ExplicitPathsOptions)</code><br/> This method will be removed in the next release. Use IRepository.Remove() instead. Obsoleted with warning.</li>
  
  <li><code>void Remove(IEnumerable&lt;string&gt;, bool, LibGit2Sharp.ExplicitPathsOptions)</code><br/> This method will be removed in the next release. Use IRepository.Remove() instead. Obsoleted with warning.</li>
  
  <li><code>void Reset(LibGit2Sharp.Tree)</code><br/> This method will be removed in the next release. Use Replace instead. Obsoleted with warning.</li>
  
  <li><code>LibGit2Sharp.FileStatus RetrieveStatus(string)</code><br/> This method will be removed in the next release. Use IRepository.RetrieveStatus() instead. Obsoleted with warning.</li>
  
  <li><code>LibGit2Sharp.RepositoryStatus RetrieveStatus(LibGit2Sharp.StatusOptions)</code><br/> This method will be removed in the next release. Use IRepository.RetrieveStatus() instead. Obsoleted with warning.</li>
  
</ul>



<h3>LibGit2Sharp.ReferenceCollection</h3>






<h4>Obsolete Methods</h4>
<ul>
  
  <li><code>LibGit2Sharp.Reference Move(LibGit2Sharp.Reference, string, LibGit2Sharp.Signature, string, bool)</code><br/> This method will be removed in the next release. Please use Rename() instead. Obsoleted with warning.</li>
  
  <li><code>LibGit2Sharp.Reference Move(LibGit2Sharp.Reference, string, bool)</code><br/> This method will be removed in the next release. Please use Rename() instead. Obsoleted with warning.</li>
  
</ul>



<h3>LibGit2Sharp.Repository</h3>






<h4>Obsolete Methods</h4>
<ul>
  
  <li><code>string get_Version()</code><br/> This property will be removed in the next release. Use GlobalSettings.Version instead. Obsoleted with warning.</li>
  
</ul>





    


